### PR TITLE
Prep for lstbin subpackage

### PR DESCRIPTION
**Pull request by steven-murray**
Adds a new "type" of makeflow creator, for the LSTbin-notebook.

### CODE REVIEW

File: hera_opm/mf_tools.py [hunk 1]
```diff
@@ -704,6 +709,22 @@ def build_makeflow_from_config(
     return
 
 
+def _get_timeout(config):
```
Comment (plaplant) [inline hera_opm/mf_tools.py:712]: Smart to pull this out to a separate function 👍

File: hera_opm/mf_tools.py [hunk 2]
```diff
@@ -1391,7 +1486,6 @@ def build_lstbin_makeflow_from_config(
 
     """
     # import hera_cal
```
Comment (plaplant) [inline hera_opm/mf_tools.py:1488]: If we're going to remove the `hera_cal` import, let's get rid of this comment too.

File: hera_opm/mf_tools.py [hunk 3]
```diff
@@ -1612,6 +1641,189 @@ def build_lstbin_makeflow_from_config(
     return
 
 
+def build_lstbin_notebook_makeflow_from_config(
+    config_file: str | Path,
```
Comment (plaplant) [inline hera_opm/mf_tools.py:1645]: Big fan of adding the type hinting here, thanks for that.

**Review by plaplant — COMMENTED** [review]
Thanks very much @steven-murray! This looks great.

One general comment I have is that I'm a little worried about the large dip in test coverage. I know that sometimes it can't be helped, especially if it depends on having different options in the config files, but this seems like a significant number of lines that are not covered by tests (almost 100 lines by codecov's reckoning). How hard would it be to increase the coverage for some of these lines?

If it's not easy to increase coverage, and we've tested that these new functions work "in the wild", then we can merge as is. Just checking to see how big a lift it would be.

**Comment by steven-murray** [comment]
@plaplant OK I thought through it a bit and realized that I was just trying to support too many different iterations of hera_cal. I've updated it so that it now only supports the newer interface -- if we need to re-run older makeflows, we will just need to install older versions of `hera_opm`, which should be perfectly easy to do. I also realized that with the new configuration in hera_cal, the creation of notebook-based and script-based makeflows is very similar, so I merged them into one function. Coverage is back up to very high.

If you think we need to maintain support for older analysis configurations, then I can try putting those back in.

**Comment by jsdillon** [comment]
What's the status of this PR? Apparently, it's necessary for the new LST-binner (and therefore H6C analysis).

**Comment by steven-murray** [comment]
@plaplant this is now ready I think. the code coverage is not 100% but the lines missed are mostly exceptions for corner cases that don't occur naturally.

**Comment by jsdillon** [comment]
Is this ready to go @plaplant ? I'd love to stop using a branch for H6C IDR3.2

File: hera_opm/tests/test_lstbin_makeflow.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        mf_name=outfile.name if give_mf_name else None,
+        work_dir=work_dir,
+        outdir=work_dir,  # pass directly so that we can check the output
+    )
+
+    # make sure the output files we expected appeared
+    assert outfile.exists()
+
+
+# @hc_skip
```
[the remark this answers is not in the document]
Comment (jsdillon) [reply hera_opm/tests/test_lstbin_makeflow.py]: I've done this

**Review by jsdillon — APPROVED** [review]
